### PR TITLE
Add batch norm option to LSTM.

### PR DESCRIPTION
**Pull request by allanzelener**
Implements "Recurrent Batch Normalization" by Cooijmans et al. https://arxiv.org/abs/1603.09025

Some key details and outstanding issues:
- Recommended initialization for (gamma, beta) parameters is provided. Gamma needs to be < 1 for batch norm to really work in the recurrent regime. See the paper for details
- (Gamma, beta) are shared through time but each time step requires independent running avg and var estimates. This implementation uses a fixed max number of batch norm layers and reuses the last one if needed for long sequences. This can also be made adaptive during training, but I'm not sure how safe it is to add layers mid-training in autograd.
- nn.BatchNormalization layers maintain avg and var estimates as separate state that can't easily be passed to an autograd function with optimize=true. Just trying to capture layers as an upvalue doesn't seem to work with optimize=true.

### CODE REVIEW

**Comment by alexbw** [comment]
Tests failing, looks like module isn't importing correctly

**Comment by alexbw** [comment]
LGTM!
